ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java

```diff
@@ -203,18 +235,22 @@ private List<Wei> computeRewards(final List<Double> rewardPercentiles, final Blo
     final ArrayList<Wei> rewards = new ArrayList<>();
     int rewardPercentileIndex = 0;
     long gasUsed = 0;
-    for (final Map.Entry<Transaction, Long> transactionAndGasUsed :
+    for (final TransactionInfo transactionAndGasUsed :
         transactionsAndGasUsedAscendingEffectiveGasFee) {
 
-      gasUsed += transactionAndGasUsed.getValue();
+      gasUsed += transactionAndGasUsed.gasUsed();
 
       while (rewardPercentileIndex < rewardPercentiles.size()
           && 100.0 * gasUsed / block.getHeader().getGasUsed()
               >= rewardPercentiles.get(rewardPercentileIndex)) {
-        rewards.add(transactionAndGasUsed.getKey().getEffectivePriorityFeePerGas(baseFee));
+        rewards.add(transactionAndGasUsed.effectivePriorityFeePerGas);
         rewardPercentileIndex++;
       }
     }
+    // Put the computed rewards in the cache
+    RewardCacheKey key = new RewardCacheKey(block.getHeader().getBlockHash(), rewardPercentiles);
```

Comment:
maybe merge this two lines into one line ?

Reply:
Done